### PR TITLE
[Component]: Label

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Adds a Label component
Fixes https://github.com/brave/leo/issues/63

~~**Note:** The hero color doesn't work because it requires #75~~

### CODE REVIEW

**Comment by fallaciousreasoning** [comment]
So I was just having a look at our variables - it looks like the font-weight for `--font-text-x-small-regular` is `400` and we don't have any separate variables for `font-weight`. Do you think we should add some?

Also, I don't know if we can (or should) have separate styles for Apple devices. I had a quick look and it doesn't seem like there's any easy way of doing it. @petemill do you have any suggestions?

**Comment by AlanBreck** [comment]
I would agree that it's best if we can avoid having device specific styling. One thing we [_could_](https://www.youtube.com/watch?v=g3j9muCo4o0) do is to add CSS rules for things like `.is-macos` or `.is-ios`, and then leave the actual OS detection up to the client, and inform them that they can add these classes (like `.is-macos`) to the body.

File: src/components/label/label.svelte [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+<div class="leo-label" style:background style:color={text}>
+  <div class="content">
+    <slot>Label</slot>
+  </div>
+</div>
+
+<style lang="scss">
+  .leo-label {
+    --icon-size: var(--leo-label-icon-size, 13px);
+    --font-text: var(--leo-label-font-text, var(--font-text-x-small-semibold));
```
Comment (petemill) [inline src/components/label/label.svelte:48]: The figma now uses `--font-components-label`

File: src/components/label/label.svelte [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    'secondary',
+    'primary',
+    'red',
+    'yellow',
+    'green',
+    'blue',
+    'purple',
+    'pink',
+    'teal',
+    'hero'
```
Comment (petemill) [inline src/components/label/label.svelte:13]: `hero` as a color should actually be `CTA`. `heavy` mode should be called `hero`

File: src/components/label/label.svelte [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  ] as const
+  export type Color = typeof colors[number]
+
+  export const modes = ['faint', 'heavy'] as const
```
Comment (petemill) [inline src/components/label/label.svelte:17]: `modes` should be renamed to `'loud' | 'default'` cc @aguscruiz

**Review by petemill — APPROVED** [review]

File: src/components/label/label.svelte (hunk 2 above)
Comment (fallaciousreasoning) [reply src/components/label/label.svelte:13]: Renamed to `CTA`

File: src/components/label/label.svelte (hunk 3 above)
Comment (fallaciousreasoning) [reply src/components/label/label.svelte:17]: Renamed - defaults to `default`

**Pull request merged**